src/stream/Middleware.ts

```diff
@@ -169,6 +180,11 @@ export class OrderedHistoryResumer extends MiddlewareBase {
     // This is sufficiently low likelihood that this can be ignored for now.
     this.historicalMessages.sort((a, b) => this.reverseOrderer(a.id, b.id));
 
+    if (this.sequenceId === '0') {
+      // there will not be a message with ID = 0 so we don't bother searching through the history
+      return false;
+    }
```

Comment:
If it is true, then I think with seq = 0, we always hit a branch in this method where we exit immediately. So I wonder if we should do that check aboce `addHistoricalMessages(...)`. 

This suggests that that: 
- In all cases where seq=0, there will never be useful history. If so, there's no point even looking at the history?

Reply:
So I guess its not strictly true just unlikely given that ADBC will produce messages that start with an ID of 1 due to the serial type on the 'id' column of the outbox table. I was trying to save us doing searches because after addHistoricalMessages is called we check the value of the sequenceId again and if its '0' we apply what ever history we have. 

We do this because '0' well always be lowest ID even if it exists the history it will still be applied. Thinking about it now I guess this is also an assumption so I could remove this check for now.

For context the solution to handling a sequenceId of zero is that we reply what ever history we have as seeking to a message with '0' id will likely never find anything and we assume a state with '0' is one that has never had any messages applied.

This is outline in the ticket comments: https://ably.atlassian.net/browse/DTP-711